gpusim.cpp

```diff
@@ -40,6 +42,43 @@ const int DATABASE_VERSION = 3;
 
 namespace gpusim
 {
+
+class DecompressAssignFPRunnable : public QRunnable {
+public:
+    QByteArray compressed_data;
+    vector<char>& fp_vector;
+
+    DecompressAssignFPRunnable(vector<char>& fp_in) : fp_vector(fp_in) {};
+    void run() override {
+        QByteArray fp_qba = qUncompress(compressed_data);
+        compressed_data.clear();
```

Comment:
why clear this out? If you ran things twice, wouldn't that assign and empty array to fp_vector?

Reply:
You would never run it twice, once you decompress the data you're done with the compressed data you read.  This removes the compressed data from taking up memory after this point.